src/Features/Core/Portable/CodeFixes/AddExplicitCast/InheritanceDistanceComparer.cs

```diff
@@ -30,63 +32,75 @@ namespace Microsoft.CodeAnalysis.CSharp.CodeFixes.AddExplicitCast
     /// 
     /// 'Derived1' is less specific than 'Derived2' compared to 'Base'
     /// </summary>
-    internal sealed class InheritanceDistanceComparer : IComparer<ITypeSymbol?>
+    internal sealed class InheritanceDistanceComparer<
+        TExpressionSyntax> : IComparer<Tuple<TExpressionSyntax, ITypeSymbol>>
+        where TExpressionSyntax : SyntaxNode
     {
-        private readonly ITypeSymbol _baseType;
         private readonly SemanticModel _semanticModel;
 
-        public InheritanceDistanceComparer(SemanticModel semanticModel, ITypeSymbol baseType)
+        public InheritanceDistanceComparer(SemanticModel semanticModel)
         {
             _semanticModel = semanticModel;
-            _baseType = baseType;
         }
 
-        public int Compare(ITypeSymbol? x, ITypeSymbol? y)
+        public int Compare(Tuple<TExpressionSyntax, ITypeSymbol>? x, Tuple<TExpressionSyntax, ITypeSymbol>? y)
```

Comment:
```suggestion
        public int Compare((TExpressionSyntax syntax, ITypeSymbol symbol)? x, (TExpressionSyntax syntax, ITypeSymbol symbol)? y)
```

Then you can use `x.symbol` or `x.syntax` instead of `x.Item2` and `x.Item1`